src/operationresult.h

```diff
@@ -27,4 +28,18 @@ inline OperationResult errorOut(const std::string& errorStr)
     return OperationResult(false, errorStr);
 }
 
+
+template <class T>
+class CallResult : public OperationResult {
+private:
+    Optional<T> m_obj_res{nullopt};
+public:
+    CallResult(bool _res, const std::string& _error) : OperationResult(_res, _error) { }
+    CallResult(bool _res, T _obj) : OperationResult(_res), m_obj_res(_obj) { }
+    CallResult(T _obj) : OperationResult(true), m_obj_res(_obj) { }
+    CallResult(const std::string& error) : OperationResult(false, error) { }
```

Comment:
It will not compile, the compiler will throw an ambiguous constructor error if you try to initialize it.

For now, we don't need to add support for it but in the future we can easily add different CallResult factory functions and remove this two public constructors.

Reply:
yup